src/app/hook.ts

```diff
@@ -0,0 +1,5 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { AppDispatch, RootState } from './store';
+
+export const useAppSelector = useSelector.withTypes<RootState>();
+export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
```

Comment:
The `useSelector` and `useDispatch` hooks do not have a `withTypes` method. Instead, you should use TypeScript's generic syntax directly with these hooks. For example, `export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;` and `export const useAppDispatch = () => useDispatch<AppDispatch>();`.